### ISSUE TITLE
ability to use network download

### ISSUE THREAD

**Issue by ramonskie** [pivotal-cf/om#306]
it would be nice if the om cli would be able to use the pivotal network download
https://docs.pivotal.io/pivotalcf/2-2/opsman-api/#download-a-given-product-with-version-from-pivotal-network

**Comment by aegershman** [comment]
Just to clarify-- instead of downloading the pivnet file externally & uploading it into the opsmgr, you'd use `om` to tell opsmgr itself to go download the pivnet file? Not trying to hijack, just workshopping out some thoughts here:

I imagine the added value as:
- If opsmgr already has the pivnet product, `om` could halt immediately. Even though `om` halts when it uploads a product and opsmgr already has it (e.g., it doesn't "double upload"), you still have to download it first. This could remove the download step entirely.
- Could this also be used for stemcells?

Problems I see:
- Moot point for replicated tiles, e.g. isolation segments (this doesn't apply to everyone, of course)
- Is there a way to ensure the download finishes? E.g., if you tell opsmgr to go download the product, will it give you a way to determine when the downloaded succeeded or failed?

**Comment by ramonskie** [comment]
@aegershman 
so the added value is indeed that it doesn't download it first localy and then upload but tell opsman to download the tile from pivotal network.

the opsman api it self gives a status back. if the tile queued, downloading, completed etc
so the opsman cli should generate a progress bar just like we are doing currently with uploading a tile.

replicated tiles stays a special as always.
i haven't looked in to stemcells yet.

**Comment by xyloman** [comment]
I think this works except for the case of air gaped environments but they are special as always so they do not apply.

**Comment by ramonskie** [comment]
also this should be addition not a replacement of the current way of uploading tiles

**Comment by kcboyle** [comment]
I hear you, and we appreciate the feature request.

We've looked into this issue ourselves when we implemented `download-product`, and we found that, while the Ops Manager API endpoint existed, it's stability across all of the IaaSs was iffy at best, and opted to do it another way. We also looked into the EULA, and found problems with allowing people to download from pivnet (unless you had an `@pivotal.io` email address). 

Is this something you were able to get working? If so we'd be happy to review and accept a PR

**Comment by fredwangwang** [comment]
background: https://pivotal.slack.com/archives/C0554UNSS/p1524587267000551?thread_ts=1524174568.000470&cid=C0554UNSS (not sure if you can see)

TLDR: the OpsManager does not support download from Pivnet if the tile has never been uploaded before. They said there is no plan to prioritize the story since a good portion of the foundations are air-gapped. (Convo from several month ago, not sure the current status).

With that been said, the api may not provide that much value, even for online foundations, as you still need to download & upload for the first time. Also, last time I tried to use it, it is not stable... network connection is kinda flaky

**Comment by kcboyle** [comment]
We're going to close this issue based on our knowledge of the limitations of the download-product feature offered in Ops Manager. If this changes, feel free to reopen this issue with a comment.
